internal/provider/resource_cloudflare_pages_domain.go

```diff
@@ -78,3 +80,39 @@ func resourceCloudflarePagesDomainDelete(ctx context.Context, d *schema.Resource
 	}
 	return nil
 }
+
+func resourceCloudflarePagesDomainImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(*cloudflare.API)
+
+	// split the id so we can look up
+	idAttr := strings.SplitN(d.Id(), "/", 2)
+	var accountID string
+	var projectName string
+	if len(idAttr) == 2 {
+		accountID = idAttr[0]
+		projectName = idAttr[1]
+	} else {
+		return nil, fmt.Errorf("invalid id %q specified, should be in format \"accountID/project-name\" for import", d.Id())
+	}
+
+	domains, err := client.GetPagesDomains(ctx, cloudflare.PagesDomainsParameters{AccountID: accountID, ProjectName: projectName})
+	if err != nil {
+		return nil, fmt.Errorf("unable to find domains for project %q: %w", d.Id(), err)
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("Found domains: %+v", domains))
+
+	resourceData := []*schema.ResourceData{}
+
+	for _, domain := range domains {
+		d.SetId(domain.ID)
```

Comment:
i don't think this is going to work the way we expect it to here. `Import` is only called expected to be called once per resource and only save that state (once for `example.com` and once for `example.net`, for example). here, it looks like we are expecting multiple domains and overwriting `ID` each time meaning only the last one will actually save. the resource schema and docs implies this is singular as well which is confusing here.

Reply:
i think you want [`GetPagesDomain`](https://github.com/cloudflare/cloudflare-go/blob/e2c82ef2466084055e6706b1104a7875038cf2a6/pages_domain.go#L86) here and to do it for each resource individually.

Reply:
That makes sense, got it :)